AlbumsNetCoreNoFramework/Program.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.Net;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+var http = new HttpListener();
+
+var host = "127.0.0.1";
+var port = 5000;
+var contentType = "application/json";
+
+http.Prefixes.Add($"http://{host}:{port}/");
+http.Start();
+
+var myAlbums = new[]
+{
+    new Album("1", "Pawn Hearts", "Van der Graaf Generator", 26.99),
+    new Album("2", "A Passion Play", "Jethro Tull", 17.99),
+    new Album("3", "Tales from Topographic Oceans", "Yes", 32.99)
+};
+
+// listen
+while(http.IsListening)
+{
+    var context = await http.GetContextAsync();
+
+    var responseString = "";
+    if (context.Request.RawUrl.StartsWith("/albums", StringComparison.OrdinalIgnoreCase))
+    {
+        // POST
+        if (context.Request.HttpMethod.Equals("POST",  StringComparison.OrdinalIgnoreCase))
+        {
+            // read content
+            Album album;
+            using (var reader = new StreamReader(context.Request.InputStream, context.Request.ContentEncoding))
+            {
+                album = (Album)JsonSerializer.Deserialize(reader.ReadToEnd(), typeof(Album), AlbumJsonSerializerContext.Default);
+            }
```

Comment:
More efficient and async:

```suggestion
            var album = JsonSerializer.DeserializeAsync<Album>(context.Request.InputStream, AlbumJsonSerializerContext.Default);
```

Reply:
@eerhardt are the DesrializeAsync<Type> methods available for source generation?  DeserializeAsync(stream) is and I have changed to that, but I still need to unbox

Reply:
scratch that, I think I see what I was doing wrong.